File: 1341-split-a-string-in-balanced-strings/1341-split-a-string-in-balanced-strings.cpp
class Solution {
public:
    int balancedStringSplit(string s) {
        int stringcount = 0;
        int rcount = 0;
        int lcount = 0;

        for(int i = 0; i < s.length(); i++)
        {
            if(s[i] == 'R')
            {
                rcount++;
            }
            if(s[i] == 'L')
            {
                lcount++;
            }
            if(rcount == lcount)
            {
                rcount = 0;
                lcount = 0;
                stringcount++;
            }
        }

        return stringcount;
        
    }
};

File: 1341-split-a-string-in-balanced-strings/1341-split-a-string-in-balanced-strings.java
class Solution {
    public int balancedStringSplit(String s) {
        int str_count = 0;
        int cnt = 0;

        for(int i = 0; i < s.length(); i++)
        {
        //    cnt = s.charAt(i) == 'R' ? ++cnt : --cnt; // use preincrement

            if(s.charAt(i) == 'R')
            {
                cnt++;
            }
            else
            {
                cnt--;
            }

            if(cnt == 0)
            {
               
                str_count++;
            }
        }

        return str_count;
    }
}